code/src/components/Todo.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import styled from 'styled-components';
+import { useDispatch } from 'react-redux';
+import { todos } from 'reducers/todos';
+
+const moment = require('moment');
+
+const StyledTodo = styled.li`
+  display: grid;
+  grid-template-columns: 50px 1fr 40px 40px;
+  grid-gap: 0.5rem;
+  align-items: center;
+  border-top: 1px solid rgba(0, 0, 0, 0.1);
+  padding: 1rem 1rem 1rem 0;
+
+  &:first-of-type {
+    border-top: none;
+  }
+
+  &:last-of-type {
+    border-bottom: none;
+  }
+
+  &:hover {
+    cursor: pointer;
+    background-color: rgba(0, 0, 0, 0.05);
+    transition: background-color 150ms ease-in;
+  }
+
+  @media screen and (min-width: 768px) {
+    grid-template-columns: 50px 1fr 160px 40px 40px;
+  }
+`;
+
+const CheckboxWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const Checkbox = styled.input.attrs({ type: 'checkbox' })``;
+
+const TextWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  height: 100%;
+  color: ${props => (props.completed ? 'rgba(0,0,0, 0.3)' : '')};
+  font-size: 1rem;
+
+  span {
+    word-break: break-all;
+  }
+`;
+
+const TimeWrapper = styled.div`
+  display: none;
+  justify-content: flex-end;
+  align-items: center;
+  font-size: 0.7rem;
+  font-style: italic;
+  margin-right: 0.9rem;
+  color: ${props => (props.completed ? 'rgba(0,0,0, 0.3)' : '')};
+
+  @media screen and (min-width: 768px) {
+    display: flex;
+  }
+`;
+
+const DeleteWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const DeleteButton = styled.button`
+  background-color: transparent;
+  color: rgba(0, 0, 0, 0.2);
+  border: none;
+  font-size: 1.1rem;
+  opacity: ${props => (props.completed ? 0.3 : 1)};
+
+  &:hover {
+    color: rgba(230, 0, 60, 1);
+    opacity: 1;
+    cursor: pointer;
+    transition: all 200ms ease-in-out;
+  }
+`;
+
+const PinWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const PinButton = styled.button`
+  background-color: transparent;
+  color: ${props =>
+    props.pinned ? 'rgba(255, 191, 0, 1)' : 'rgba(0, 0, 0, 0.2)'};
+  border: none;
+  font-size: 1.1rem;
+  opacity: ${props => (props.completed ? 0.3 : 1)};
+
+  &:hover {
+    opacity: 1;
+    color: ${props =>
+      props.pinned ? 'rgba(0, 0, 0, 0.2)' : 'rgba(255, 191, 0, 1);'};
+    cursor: pointer;
+    transition: all 200ms ease-in-out;
+  }
+`;
+
+export const Todo = ({ todo }) => {
+  const dispatch = useDispatch();
+
+  return (
+    <StyledTodo>
+      <CheckboxWrapper>
+        <Checkbox
+          onChange={() => dispatch(todos.actions.toggleCompleted({ todo }))}
```

Comment:
Nitpicking a bit here, but since this is reused so much, I think it'd be a little clearer if you made a function out of it and used that in all the `onChange` attrs. 

```js
const toggleCompleted = () => {
  dispatch(todos.actions.toggleCompleted({ todo }))
}

< Checkbox onChange={toggleCompleted} />
<TextWrapper onChange={toggleCompleted} />
//.. etc
```

... That way it's easier to see at a glance which ones are doing the same things.

Reply:
Nitpicking is totally fine! 

It's very easy to drift away from the DRY principle. But I see your point here. I could restructure things a bit and make it more readable.

I don't have to pass the todo as an argument in the toggleCompleted function?